File: app/src/main/java/com/chydee/notekeeper/dao/NoteDao.java
package com.chydee.notekeeper.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.chydee.notekeeper.model.Note;

import java.util.List;

@Dao
public interface NoteDao {

    //Insert or add a note to the note_table
    @Insert
    void insert(Note note);

    //Update a single note
    @Update
    void update(Note note);

    //Delete a single note from the note_table
    @Delete
    void delete(Note note);

    //Delete all notes inside the note_table
    @Query("DELETE FROM note_table")
    void deleteAll();

    //This method returns all the notes needed so that we can put them in our recycler view
    @Query("SELECT * FROM note_table ORDER BY priority DESC")
    LiveData<List<Note>> getAllNotes();
}

File: app/src/main/java/com/chydee/notekeeper/data/dao/NoteDao.kt
package com.chydee.notekeeper.data.dao

import androidx.room.*
import com.chydee.notekeeper.data.model.Note
import io.reactivex.Completable
import io.reactivex.Single

// Defines the method for using the Note class with room
@Dao
interface NoteDao {

    /*
    * This method inserts the note into the database
    */
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(note: Note): Completable

    @Insert
    fun insert(notes: List<Note>): Completable

    /*
    * This method updates a single note from the note database
     */
    @Update
    fun update(note: Note): Completable

    @Delete
    fun delete(note: Note): Completable

    /**
     * Deletes all values from the table.
     *
     * This does not delete the table, only its contents.
     */
    @Query("DELETE FROM note")
    fun deleteAllNotes(): Completable

    /**
     * Selects and returns the latest night.
     */
    @Query("SELECT * FROM note ORDER BY id DESC LIMIT 1")
    fun getLatestNote(): Single<Note>

    @Query("SELECT * FROM note ORDER BY id DESC")
    fun getAllNotes(): Single<List<Note>>
}
